lib/processors/versionInfoGenerator.js

```diff
@@ -34,20 +424,94 @@ module.exports = async function({options}) {
 	}
 
 	const buildTimestamp = getTimestamp();
+
+	/**
+	 * componentName to dependency info
+	 *
+	 * @type {Map<string, DependencyInfo>}
+	 */
+	const dependencyInfoMap = new Map();
+
+
+	// process library infos
+	const librariesPromises = options.libraryInfos.map((libraryInfo) => {
+		return processLibraryInfo(libraryInfo);
+	});
+
+	let artifactInfos = await Promise.all(librariesPromises);
+	artifactInfos = artifactInfos.filter(Boolean);
+
+	// fill dependencyInfoMap
+	artifactInfos.forEach((artifactInfo) => {
+		dependencyInfoMap.set(artifactInfo.componentName, artifactInfo.dependencyInfo);
+	});
+
+	// resolve library dependencies (transitive)
+	dependencyInfoMap.forEach((dependencyInfo) => {
+		dependencyInfo.resolve(dependencyInfoMap);
+	});
+
+	// resolve dependencies of embedded components
+	artifactInfos.forEach((artifactInfo) => {
+		artifactInfo.getEmbeds().forEach((embeddedArtifactInfo) => {
+			embeddedArtifactInfo.dependencyInfo.resolve(dependencyInfoMap);
+		});
+	});
+
+	const libraries = options.libraryInfos.map((libraryInfo) => {
+		const library = {
+			name: libraryInfo.name,
+			version: libraryInfo.version,
+			buildTimestamp: buildTimestamp,
+			scmRevision: ""// TODO: insert current library scm revision here
+		};
+
+		const dependencyInfo = dependencyInfoMap.get(libraryInfo.name);
+		const manifestHints = getManifestHints(dependencyInfo);
+		if (manifestHints) {
+			library.manifestHints = manifestHints;
+		}
+		return library;
+	});
+
+	// sort libraries alphabetically
+	libraries.sort((a, b) => {
+		return a.name.localeCompare(b.name);
+	});
+
+	// components
+	let components;
+	artifactInfos.forEach((artifactInfo) => {
+		artifactInfo.getEmbeds().forEach((embeddedArtifactInfo) => {
+			const componentObject = {
+				library: embeddedArtifactInfo.parentComponentName
```

Comment:
can `embeddedArtifactInfo.parentComponentName` ever differ from `artifactInfo.name`?

Reply:
tackled with fd6a8e9
Put ArtifactInfo as typedef because the parent fields in the child could be retrieved by directly using the parent's fields.